src/basicfuns.jl

```diff
@@ -407,3 +407,17 @@ function _softmax!(r, x, dims)
     end
     return r
 end
+
+"""
+$(SIGNATURES)
+
+Compute the complementary log-log, `log(-log(x))`.
```

Comment:
```suggestion
Compute the complementary log-log, `log(-log(1-x))`.
```

Reply:
Wow, this is why I shouldn't make PRs after 11pm. I was thinking of the complementary log-log in terms of its use in survival analysis, which is `log(-log(x))`, but copied the implementation for its use as a link function, which is indeed `log(-log(1 - x))`. Would it be worthwhile to have both, do you think?

Reply:
Generally, I think one should add things which have a use case. If you need both, sure, just add it, otherwise it can wait for another PR.

Reply:
My uses for both would be in Survival.jl and GLM.jl but might as well stick with what's implemented in this PR for now and do a follow-up if I can't just rearrange some stuff in Survival to use this formulation.